Course/Course/Presentation/Container/CourseContainerViewModel.swift

```diff
@@ -430,6 +468,40 @@ public class CourseContainerViewModel: BaseCourseViewModel {
                 self.isInternetAvaliable = self.connectivity.isInternetAvaliable
         }
         .store(in: &cancellables)
+        
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleShiftDueDates(_:)),
+            name: .shiftCourseDates, object: nil
+        )
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+}
+
+extension CourseContainerViewModel {
+    @objc private func handleShiftDueDates(_ notification: Notification) {
+        if let courseID = notification.object as? String {
+            Task {
+                await withTaskGroup(of: Void.self) { group in
+                    group.addTask {
+                        await self.getCourseBlocks(courseID: courseID, withProgress: true)
+                    }
+                    group.addTask {
+                        await self.getCourseDeadlineInfo(courseID: courseID, withProgress: true)
+                    }
+                    DispatchQueue.main.async {
+                        self.dueDatesShifted = true
+                    }
+                }
+            }
+        }
+    }
+    
+    func resetDueDatesShiftedFlag() {
+        dueDatesShifted = false
```

Comment:
Do we need a method just to change one variable? Can we directly change it's value as the variable isn't private.

Reply:
This variable cannot be set as private since it is a `@Published` property and is being displayed in various views. The reason for creating a method to set its default value is to ensure that the view model is responsible for updating the property value, maintaining a single source of truth.